### PR TITLE
Blink mempool replacement

### PR DESCRIPTION
**Pull request by jagerman**
This adds support for bumping conflicting transactions from the mempool
upon receipt of a blink tx if those conflicting txs are not themselves
blinks.

This commit chops the handle_incoming_txs effectively in half into a
parse_incoming_txs phase, followed by a handle_incoming_txs phase which
is needed so that we can first parse the txes then, assuming they parse
successfully, can flag them as blink txes if blink info was included
with them, so that the actual insertion becomes forcing.

There is a stub here for also allowing a rollback of the mutable part of
the chain; implementation of actually doing that rollback will follow in
another PR.

~~Note: there is still an issue here with how txes are being stored in the database;~~ and the test suite still needs stubs added to compile properly.  Will follow up with a fix for both, but this is ready for review currently.

### CODE REVIEW

**Comment by jagerman** [comment]
(This builds on the other PRs, starting after `Merge branch 'blink-fees' into blink-replace`)

File: src/cryptonote_core/tx_pool.cpp [hunk 1]
```diff
@@ -586,109 +586,110 @@ namespace cryptonote
...
         // don't prune the kept_by_block ones, they're likely added because we're adding a block with those
         if (meta.kept_by_block)
-        {
-          it++;
-          continue;
-        }
+          return true;
 
-        if (!prune_tx(it, txid, meta, changed))
+        if (remove_tx(txid, &meta, &del_it))
```
[the remark this answers is not in the document]
Comment (jagerman) [reply src/cryptonote_core/tx_pool.cpp]: `del_it` comes from: `auto del_it = forward ? it++ : it--;` a couple lines earlier so `it` has already been moved to the next element; whatever happens with `del_it`, `it` pointing at the next value will stay valid.

[the remark this answers is not in the document]
Comment (Doy-lee) [reply src/cryptonote_core/tx_pool.cpp]: Oh yeah but doesn't `del_it` delete the wrong element? If you pass in `it` into `try_pruning` then `auto del_it = forward ? it++ : it--` change what I'm deleting to +-1 of the actual element I intended to remove or am I missing something.

[the remark this answers is not in the document]
Comment (jagerman) [reply src/cryptonote_core/tx_pool.cpp]: It's a post-increment, so `it` gets incremented and we get back whatever `it` was pointing to before the increment, which is the one we tested above.

[the remark this answers is not in the document]
Comment (jagerman) [reply src/cryptonote_core/tx_pool.cpp]: I would normally much prefer using a `it = container.remove(it)`, as I think it's clearer in intent, but that doesn't work for the common case here where we're iterating in reverse.

[the remark this answers is not in the document]
Comment (Doy-lee) [reply src/cryptonote_core/tx_pool.cpp]: Hmmm ok. It checks out

**Pull request merged**